openid_configuration.rb

```diff
@@ -27,6 +27,18 @@ def self.live
           end
         end
       end
+
+      def self.cached_idp_public_key(openid_configuration)
+        @idp_public_key ||= live_idp_public_key(openid_configuration)
```

Comment:
can we give this cache a lifetime limit? if the server never restarts, the cache will not expire, right?

Maybe something like https://github.com/derrickreimer/mini_cache

We set cache expiration headers on the response so we can use those to inform the cache lifetime?

Reply:
I was mostly relying on cloud.gov restarting arbitrarily ([docs](https://cloud.gov/docs/deployment/deployment/#the-platform-will-restart-your-application)):

>Application instances will also be restarted when the cloud.gov platform is updated, which can be several times a week. This shouldn’t disrupt your running application if you set up multiple application instances and avoid writing to the local file system.

It looks like the IDP sends 1 week cache expiration headers regardless, so I don't know how useful the expiration is, though the response is ETagged. I think the bigger issue is the example app always takes the first key and doesn't iterate through if there were multiple keys.

Reply:
That works too 👍